qiskit_experiments/library/characterization/analysis/drag_analysis.py

```diff
@@ -221,36 +181,69 @@ def _run_curve_fit(
 
         Args:
             curve_data: Formatted data to fit.
-            series: A list of fit models.
+            models: A list of LMFIT models that are used to build a cost function
+                for the LMFIT minimizer.
 
         Returns:
             The best fitting outcome with minimum reduced chi-squared value.
         """
-        fit_result = super()._run_curve_fit(curve_data, series)
-        beta = fit_result.popt[2]
-        freq = fit_result.popt[1]
-        fit_result.popt[2] = ((beta + 1 / freq / 2) % (1 / freq)) - 1 / freq / 2
+        fit_result = super()._run_curve_fit(curve_data, models)
+
+        if fit_result:
+            # covert into dict since dataclass is frozen
+            data_dict = fit_result.__json_encode__()
+            params = data_dict["params"]
+            beta = params["beta"]
+            freq = params["freq"]
+            min_beta = ((beta + 1 / freq / 2) % (1 / freq)) - 1 / freq / 2
+            data_dict["params"]["beta"] = min_beta
+            return curve.CurveFitResult.__json_decode__(data_dict)
 
         return fit_result
 
-    def _evaluate_quality(self, fit_data: curve.FitData) -> Union[str, None]:
+    def _evaluate_quality(self, fit_data: curve.CurveFitResult) -> Union[str, None]:
         """Algorithmic criteria for whether the fit is good or bad.
 
         A good fit has:
             - a reduced chi-squared lower than three,
             - a DRAG parameter value within the first period of the lowest number of repetitions,
             - an error on the drag beta smaller than the beta.
         """
-        fit_beta = fit_data.fitval("beta")
-        fit_freq = fit_data.fitval("freq")
+        fit_beta = fit_data.ufloat_params["beta"]
+        fit_freq = fit_data.ufloat_params["freq"]
 
         criteria = [
             fit_data.reduced_chisq < 3,
             abs(fit_beta.nominal_value) < 1 / fit_freq.nominal_value / 2,
-            curve.is_error_not_significant(fit_beta),
+            curve.utils.is_error_not_significant(fit_beta),
         ]
 
         if all(criteria):
             return "good"
 
         return "bad"
+
+    def _initialize(
+        self,
+        experiment_data: ExperimentData,
+    ):
+        super()._initialize(experiment_data)
+
+        # Model is initialized at runtime because
+        # the experiment option "reps" can be changed before experiment run.
+        plot_options = {}
+        for i, nrep in enumerate(sorted(self.options.reps)):
+            name = f"nrep={nrep}"
+            self._models.append(
+                ExpressionModel(
+                    expr=f"amp * cos(2 * pi * {nrep} * freq * (x - beta)) + base",
+                    name=name,
+                    data_sort_key={"nrep": nrep},
+                )
+            )
+            plot_options[name] = {
+                "color": curve.utils.colors10(i),
+                "symbol": curve.utils.symbols10(i),
```

Comment:
Can this be automated by a plot style like matplotlib/seaborn do and we can specify our own plot style rather than these util functions? eg:
* https://seaborn.pydata.org/generated/seaborn.set_style.html
* https://matplotlib.org/stable/gallery/style_sheets/style_sheets_reference.html

Reply:
Good idea. The point here is we should set apply the same property for the same curve (e.g. data point plot and fit line plot for the same curve use the same color). This is implemented based on matplotlib default `tab10` color cycle. I cannot find cycler for markers.